src/movie/Edit.js

```diff
@@ -0,0 +1,183 @@
+import React, { useReducer, useState } from 'react';
+import { useParams, useHistory } from 'react-router';
+import { Link } from 'react-router-dom';
+import { Alert } from 'react-bootstrap';
+import { movieList, genres } from './Const';
+
+import './Edit.scss';
+
+export function EditMovie() {
+  const [formErrorShow, setFormErrorShow] = useState(false);
+  const [submitSuccessShow, setSubmitSuccessShow] = useState(false);
+  const history = useHistory();
+
+  let { id } = useParams();
+  id = Number(id) || 0;
+  const isAddMovie = !id;
+  const movieDetails = isAddMovie ? {} : movieList.find((movie) => movie.id === id);
+
+  const formReducer = (movieFormState, newMovieFormState) => ({
+    ...movieFormState,
+    ...newMovieFormState,
+  });
+  const [movieFormState, dispatchInputChange] = useReducer(
+    formReducer,
+    movieDetails
+  );
+
+  const getMultiSelectValue = (element) => {
+    const selectedOptions = Array.prototype.slice.call(element.selectedOptions);
```

Comment:
`element.selectedOptions` is a HTMLCollection type instead of array hence I had to first convert it to array. I have replaced `slice` with spread operator now.